compiler/rustc_middle/src/ty/query/on_disk_cache.rs

```diff
@@ -575,25 +574,23 @@ impl<'sess> OnDiskCache<'sess> {
     // maps to None.
     fn compute_cnum_map(
         tcx: TyCtxt<'_>,
-        prev_cnums: &[(u32, String, CrateDisambiguator)],
+        prev_cnums: &[(u32, StableCrateId)],
     ) -> IndexVec<CrateNum, Option<CrateNum>> {
         tcx.dep_graph.with_ignore(|| {
             let current_cnums = tcx
                 .all_crate_nums(())
                 .iter()
                 .map(|&cnum| {
-                    let crate_name = tcx.original_crate_name(cnum).to_string();
-                    let crate_disambiguator = tcx.crate_disambiguator(cnum);
-                    ((crate_name, crate_disambiguator), cnum)
+                    let stable_crate_id = tcx.def_path_hash(cnum.as_def_id()).stable_crate_id();
```

Comment:
Avoiding the `def_path_hash` call here in favor of directly looking up the `StableCrateId` by `CrateNum` from the `CStore` may also help.